### PR TITLE
feat(confgen): auto deduce whether the map key is unique

### LINKED ISSUE #78: confgen: auto deduce whether the map key is unique
By default, map key is not treated unique, in order to aggregate sub-field
(map or list) with cardinality. The map field key should be deduced to be
unique if nesting hierarchy is:

  - map nesting map or list with different layout (vertical or horizontal).
  - map nesting no map or list.
  - map layout is incell.

### PR DESCRIPTION
**Pull request by Kybxd**
- close #78

### CODE REVIEW

File: internal/localizer/i18n/config/ecode/zh.yaml [hunk 1]
```diff
@@ -80,6 +80,10 @@ E2016:
   desc: list elements are not present continuously
   text: 数组第{{.FirstNonePresentIndex}}个元素为空, 但后续第{{.NextPresentIndex}}个元素却非空
   help: 请确保数组出现首个空元素时, 后续所有元素都为空
+E2017:
+  desc: map contains multiple empty keys
+  text: map "{{.Name}}" 存在超过一个的key为空
+  help: 请确保map中最多仅有一个key为空, 如果确实需要map中多个key为零值请手动填充零值
```
Comment (wenchy) [inline internal/localizer/i18n/config/ecode/zh.yaml:86]: change to:
```
  desc: map contains multiple empty keys
  text: map "{{.Name}}" 多个空值key
  help: 请确保map中最多仅有一个空值key，如果确实需要多个key为零值，请显式填充。
```

**Review by wenchy — CHANGES_REQUESTED** [review]

File: internal/localizer/i18n/config/ecode/en.yaml [hunk 2]
```diff
@@ -80,6 +80,10 @@ E2016:
   desc: list elements are not present continuously
   text: The {{.FirstNonePresentIndex}}-th element is not present, but the subsequent {{.NextPresentIndex}}-th element is present
   help: ensure all subsequent elements after the first none-present element are not present
+E2017:
+  desc: map contains multiple empty or keys
+  text: map "{{.Name}}" contains multiple empty keys
+  help: ensure there is only one empty or key in map, if you need multiple zero-value keys you should explicitly assign a zero value
```
[the remark this answers is not in the document]
Comment (Kybxd) [reply internal/localizer/i18n/config/ecode/en.yaml]: done

File: internal/localizer/i18n/config/ecode/zh.yaml (hunk 1 above)
Comment (Kybxd) [reply internal/localizer/i18n/config/ecode/zh.yaml]: done

File: xerrors/ecode.go [hunk 3]
```diff
@@ -174,6 +174,11 @@ func E2016(firstNonePresentIndex, nextPresentIndex int) error {
 	})
 }
 
+// E2017: map value present while key not present.
+func E2017() error {
+	return renderEcode("E2017", nil)
```
Comment (wenchy) [inline xerrors/ecode.go]: Need variable `Name` to be injected.

**Review by wenchy — CHANGES_REQUESTED** [review]

File: xerrors/ecode.go (hunk 3 above)
Comment (Kybxd) [reply xerrors/ecode.go]: done

File: xerrors/ecode.go [hunk 4]
```diff
@@ -174,6 +174,13 @@ func E2016(firstNonePresentIndex, nextPresentIndex int) error {
 	})
 }
 
+// E2017: map value present while key not present.
```
Comment (wenchy) [inline xerrors/ecode.go:177]: change to : `// E2017: map contains multiple empty keys.`

**Review by wenchy — CHANGES_REQUESTED** [review]

File: internal/confgen/parser.go [hunk 5]
```diff
@@ -407,13 +407,22 @@ func (sp *sheetParser) parseMapField(field *Field, msg protoreflect.Message, rc
...
+				tempCheckMapValue := reflectMap.NewValue()
+				valuePresent, err := sp.parseFieldOptions(tempCheckMapValue.Message(), rc, prefix+field.opts.Name)
+				if err != nil {
 					kvs := append(rc.CellDebugKV(keyColName), xerrors.KeyPBFieldType, "vertical map")
-					return false, xerrors.WrapKV(err, kvs...)
+					return false, xerrors.WithMessageKV(err, kvs...)
+				}
+				if valuePresent {
+					kvs := append(rc.CellDebugKV(keyColName), xerrors.KeyPBFieldType, "vertical map")
+					return false, xerrors.WithMessageKV(xerrors.E2017(keyColName), kvs...)
```
Comment (wenchy) [inline internal/confgen/parser.go:420]: Use `field.fd.FullName()` and `field.fd.Kind().String()` to fill E2017 error info.

**Review by wenchy — CHANGES_REQUESTED** [review]

**Review by wenchy — APPROVED** [review]
LGTM

**Pull request merged**